skynet/modules/ttt/s3.py

```diff
@@ -0,0 +1,66 @@
+from skynet.env import (
+    skynet_s3_access_key,
+    skynet_s3_bucket,
+    skynet_s3_endpoint,
+    skynet_s3_region,
+    skynet_s3_secret_key,
+)
+from skynet.logs import get_logger
+
+log = get_logger(__name__)
+
+
+class S3:
+    def __init__(self):
+        import boto3
+
+        try:
+            self.s3 = boto3.resource(
+                's3',
+                aws_access_key_id=skynet_s3_access_key,
+                aws_secret_access_key=skynet_s3_secret_key,
+                endpoint_url=skynet_s3_endpoint,
+                region_name=skynet_s3_region,
+            )
+        except Exception as e:
+            log.error(f'Failed to connect to S3: {e}')
+
+    def download_file(self, filename):
```

Comment:
This looks blocking which is no bueno. I'd recommend having a dedicated executor or using the current loop's to run the fs ops and turn them into async.

Reply:
This should apply for all the functions here.